### PR TITLE
Closes #13

### LINKED ISSUE #13: Standardized quest collection and service
- [x] create quests collection with the ff fields:
1. id
2. name
3. description
4. status (boolean always true for now)
5. source
6. category (object containing id, name, description, & status)
7. coreTag - true if it is uploaded

- [x] Add rules for this new collection: read=true and write=false
- [x] Prepare a csv to be used to populate this collection with data
- [x] create a new interface for quest using the above fields
- [x] create new quest service with methods:
      1. to query quests under a given category, limit to 5 results per query to reduce no. of reads [good starting reference](https://angularfirebase.com/lessons/autocomplete-search-with-angular4-and-firebase/)
      2. to save a new quest and return new quest id with coreTag=false

### PR DESCRIPTION
**Pull request by aronpadolina**
Closes #13

### CODE REVIEW

File: src/app/quests/quest.service.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    );
+  }
+
+  mapQuestData(actions: any[]): Quest[] {
+    return actions.map((a) => {
+      const data = a.payload.doc.data();
+      return <Quest>{ id: a.payload.doc.id, ...data };
+    });
+  }
+
```
Comment (gerarrd) [inline src/app/quests/quest.service.ts]: Remove unnecessary spaces.

Comment (aronpadolina) [reply src/app/quests/quest.service.ts]: Done

File: firestore.rules [hunk 2]
```diff
@@ -41,7 +41,12 @@ service cloud.firestore {
       allow read: if isSignedIn();
       allow write: if false;
     }
-
+		
+    match /quests/{id} {
+	  allow read: if true;
```
Comment (nbcp) [inline firestore.rules]: indent here too

File: firestore.rules [hunk 3]
```diff
@@ -41,7 +41,12 @@ service cloud.firestore {
...
+      allow write: if isQuestWriteAllowed();
```
Comment (nbcp) [inline firestore.rules]: lets change this to `create` then restrict both `update` and `delete` by setting them to false

**Pull request merged**